tests/middleware/test_proxy_headers.py

```diff
@@ -14,28 +17,34 @@ async def app(scope, receive, send):
 app = ProxyHeadersMiddleware(app, trusted_hosts="*")
 
 
-def test_proxy_headers():
-    client = TestClient(app)
-    headers = {"X-Forwarded-Proto": "https", "X-Forwarded-For": "1.2.3.4"}
-    response = client.get("/", headers=headers)
+@pytest.mark.asyncio
+async def test_proxy_headers():
+    async with httpx.AsyncClient(app=app, base_url="http://testserver") as client:
+        headers = {"X-Forwarded-Proto": "https", "X-Forwarded-For": "1.2.3.4"}
+        response = await client.get("/", headers=headers)
     assert response.status_code == 200
     assert response.text == "Remote: https://1.2.3.4:0"
 
 
-def test_proxy_headers_no_port():
-    client = TestClient(app)
-    headers = {"X-Forwarded-Proto": "https", "X-Forwarded-For": "1.2.3.4"}
-    response = client.get("/", headers=headers)
+@pytest.mark.asyncio
+async def test_proxy_headers_no_port():
+    async with httpx.AsyncClient(app=app, base_url="http://testserver") as client:
+        headers = {"X-Forwarded-Proto": "https", "X-Forwarded-For": "1.2.3.4"}
+        response = await client.get("/", headers=headers)
     assert response.status_code == 200
     assert response.text == "Remote: https://1.2.3.4:0"
 
 
-def test_proxy_headers_invalid_x_forwarded_for():
-    client = TestClient(app)
-    headers = {
-        "X-Forwarded-Proto": "https",
-        "X-Forwarded-For": "\xf0\xfd\xfd\xfd, 1.2.3.4",
-    }
-    response = client.get("/", headers=headers)
+@pytest.mark.asyncio
+async def test_proxy_headers_invalid_x_forwarded_for():
+    async with httpx.AsyncClient(app=app, base_url="http://testserver") as client:
+        headers = Headers(
```

Comment:
We can do this to avoid importing Headers explicitly:

```suggestion
        headers = httpx.Headers(
```

Reply:
done